libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1613,6 +1613,13 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	_operation.annotation().isPure = !modifying && *_operation.subExpression().annotation().isPure;
 	_operation.annotation().isLValue = false;
 
+	if (op == Token::Delete && subExprType->nameable() && subExprType->containsNestedMapping())
```

Comment:
With the `delete` check, I feel that `nameable` condition can be removed. But not sure :(